addons/project/static/tests/project_task_calendar.test.js

```diff
@@ -0,0 +1,54 @@
+import { expect, test, beforeEach, describe } from "@odoo/hoot";
+import { mockDate, animationFrame, advanceTime } from "@odoo/hoot-mock";
+import { click, edit, queryOne } from "@odoo/hoot-dom";
+
+import { mountView, onRpc } from "@web/../tests/web_test_helpers";
+import { clickDate } from "@web/../tests/views/calendar/calendar_test_helpers";
+
+import { defineProjectModels, ProjectTask } from "./project_models";
+
+describe.current.tags("desktop");
+defineProjectModels();
+
+beforeEach(() => {
+    mockDate("2024-01-03 12:00:00", +0);
+
+    ProjectTask._views["form, false"] = `
+        <form>
+            <field name="id"/>
+            <field name="name"/>
+            <field name="date_deadline"/>
+            <field name="planned_date_begin"/>
+        </form>
+    `;
+
+    onRpc("check_access_rights", () => true);
+});
+
+test(" test Project Task Calendar Popover with task_stage_with_state_selection widget", async () => {
+    await mountView({
+        resModel: "project.task",
+        type: "calendar",
+        arch: `
+            <calendar date_start="date_deadline" mode="month"
+                        js_class="project_task_calendar">
+	            <field name="project_id" widget="project" invisible="context.get('default_project_id', False)"/>
+	            <field name="stage_id" invisible="not project_id or not stage_id" widget="task_stage_with_state_selection"/>
+	        </calendar>
+        `,
+        context: { default_project_id: 1, default_stage_id: 1, default_state_id: "01_in_progress" },
+    });
+
+    await clickDate("2024-01-09");
+    edit("Task-1");
+    click("button.o-calendar-quick-create--create-btn");
```

Comment:
Alternatively, you can also add a record inside `ProjectTask._records`, it would be easier, IMO.

Also, we should check the state is editable when `state_readonly === false` and check it is not editable when `state_readonly === true`